code/game/objects/structures/watercloset.dm

```diff
@@ -434,6 +449,16 @@
 		C.adjustFireLoss(5)
 		to_chat(C, "<span class='danger'>The water is searing!</span>")
 
+/obj/machinery/shower/proc/check_clothes(mob/living/carbon/human/H)
+	var/result 
+	result &= (H.wear_suit && !(H.wear_suit.clothing_flags & SHOWEROKAY))
+	result &= (H.w_uniform && !(H.w_uniform.clothing_flags & SHOWEROKAY))
+	result &= (H.shoes && !(H.shoes.clothing_flags & SHOWEROKAY))
+	result &= (H.ears && !(H.ears.clothing_flags & SHOWEROKAY))
+	result &= (H.gloves && !(H.gloves.clothing_flags & SHOWEROKAY))
+	result &= (H.wear_mask && !(H.wear_mask.clothing_flags & SHOWEROKAY))
+	result &= (H.head && !(H.head.clothing_flags & SHOWEROKAY))
+	return result
```

Comment:
What if you're wearing a radiation suit, but non-showerable clothes underneath the suit

Reply:
omg pls dont make me depressed